src/modules/Packages.rb

```diff
@@ -2066,32 +2066,50 @@ def SelectProduct
       Initialize(true)
 
       if Stage.cont
-        Builtins.y2milestone("Second stage - skipping product selection")
+        log.info("Second stage - skipping product selection")
         return true
       end
 
       products = Pkg.ResolvableProperties("", :product, "")
 
-      if Builtins.size(products) == 0
-        Builtins.y2milestone("No product found on media")
+      if products.empty?
+        log.info("No product found on media")
         return true
       end
 
-      selected_products = Builtins.filter(products) do |p|
-        Ops.get(p, "status") == :selected
-      end
       # no product selected -> select them all
       ret = true
-      if Builtins.size(selected_products) == 0
-        Builtins.y2milestone("No product selected so far...")
-        Builtins.foreach(products) do |p|
-          product_name = Ops.get_string(p, "name", "")
-          if !Builtins.regexpmatch(product_name, "-migration$")
-            Builtins.y2milestone("Selecting product %1", product_name)
-            ret = Pkg.ResolvableInstall(product_name, :product) && ret
+      unless products.find_index { |p| p["status"] == :selected }
```

Comment:
Why not use `any?` (or even `none?` if you want to replace `unless` by `if`) instead of `find_index`? After all, you're not interested in the index value and `any?` express better your intention.